File: 2019/day5/day5.py
#!/usr/bin/env python3
# https://adventofcode.com/2019/day/5

import sys
from pathlib import Path

sys.path.append(Path(__file__).parent.parent.as_posix())
from intcode.Intcode import Computer


filename = ("test.txt" if sys.argv[1] == "-t" else sys.argv[1]) if len(sys.argv) > 1 else "input.txt"


computer = Computer()
computer.load_from(filename)

# part 1
computer.input.append(1)
computer.run()
print(computer.output.pop())

# part 2
computer.input.append(5)
computer.run()
print(computer.output.pop())

File: 2019/day5/day5.rs
//! [Day 5: Sunny with a Chance of Asteroids](https://adventofcode.com/2019/day/5)

use intcode::{Computer, State};

fn main() {
    let args = aoc::parse_args();

    let mut program = Computer::load(&args.input);

    let mut run = |init| {
        program.reset();
        program.push(init);
        let mut result = 0;
        while let State::Output(value) = program.run() {
            result = value;
        }
        result
    };

    println!("{}", run(1)); // part 1
    println!("{}", run(5)); // part 2
}
